parachain/runtime/kintsugi/src/dex.rs

```diff
@@ -0,0 +1,243 @@
+pub struct LoansSetup;
+
+fn add_and_activate_market(token: CurrencyId, market: loans::Market<Balance>) {
+    if let Err(err) = Loans::add_market(RuntimeOrigin::root(), token, market) {
+        log::error!("Could not add market: {:?}", err);
+    } else {
+        if let Err(err) = Loans::activate_market(RuntimeOrigin::root(), token) {
+            log::error!("Could not activate market: {:?}", err);
+        }
+    }
+}
+
+impl OnRuntimeUpgrade for LoansSetup {
+    fn on_runtime_upgrade() -> Weight {
+        add_and_activate_market(
+            CurrencyId::Token(KBTC),
+            loans::Market {
+                close_factor: Ratio::from_percent(50),
+                collateral_factor: Ratio::from_percent(63),
+                liquidation_threshold: Ratio::from_percent(67),
+                liquidate_incentive: Rate::from_inner(Rate::DIV / 100 * 110),
+                liquidate_incentive_reserved_factor: Ratio::zero(),
+                state: loans::MarketState::Pending,
+                rate_model: loans::InterestRateModel::Jump(loans::JumpModel {
+                    base_rate: Rate::zero(),
+                    jump_rate: Rate::from_inner(Rate::DIV / 100 * 5),
+                    full_rate: Rate::from_inner(Rate::DIV / 100 * 50),
+                    jump_utilization: Ratio::from_percent(90),
+                }),
+                reserve_factor: Ratio::from_percent(20),
+                supply_cap: 2000000000, // 20 KBTC
+                borrow_cap: 2000000000, // 20 KBTC
+                lend_token_id: CurrencyId::LendToken(1),
+            },
+        );
+
+        add_and_activate_market(
+            CurrencyId::Token(KSM),
+            loans::Market {
+                close_factor: Ratio::from_percent(50),
+                collateral_factor: Ratio::from_percent(54),
+                liquidation_threshold: Ratio::from_percent(61),
+                liquidate_incentive: Rate::from_inner(Rate::DIV / 100 * 110),
+                liquidate_incentive_reserved_factor: Ratio::zero(),
+                state: loans::MarketState::Pending,
+                rate_model: loans::InterestRateModel::Jump(loans::JumpModel {
+                    base_rate: Rate::zero(),
+                    jump_rate: Rate::from_inner(Rate::DIV / 100 * 15),
+                    full_rate: Rate::from_inner(Rate::DIV / 100 * 40),
+                    jump_utilization: Ratio::from_percent(90),
+                }),
+                reserve_factor: Ratio::from_percent(20),
+                supply_cap: 30000000000000000, // 30,000 KSM
+                borrow_cap: 30000000000000000, // 30,000 KSM
+                lend_token_id: CurrencyId::LendToken(2),
+            },
+        );
+
+        add_and_activate_market(
+            CurrencyId::ForeignAsset(3), // USDT
+            loans::Market {
+                close_factor: Ratio::from_percent(50),
+                collateral_factor: Ratio::from_percent(65),
+                liquidation_threshold: Ratio::from_percent(69),
```

Comment:
@philippb90 there's also a difference between the discussion post and the sheet here with 69% vs 71%. Which one is the correct one?

Reply:
The implementation is correct. For USDT the values are 65% and 69%.

Please use this notion as single source of truth: https://www.notion.so/interlay/04957682901a46619cfeba643219f7a9?v=b6ec9f7152974182ba861744b3d57017&pvs=4